fda-model/compute/simulation.cs

```diff
@@ -178,22 +218,27 @@ internal Simulation()
             }
         }
         private IPairedData BootstrapToPairedData(interfaces.IProvideRandomNumbers rp, Statistics.ContinuousDistribution dist, Int64 ordinates){
-            double[] randyPacket = rp.NextRandomSequence(dist.SampleSize);
-            IDistribution bootstrap = dist.Sample(randyPacket);
-            double[] x = new double[ordinates];
-            double[] y = new double[ordinates];
-            for(int i=0;i<ordinates; i++){
-                double val = (double) i + .5;
-                //equally spaced non-exceedance (cumulative) probabilities in increasing order
-                double prob = (val)/((double)ordinates);
-                x[i] = prob;
-
-                //y values in increasing order 
-                y[i] = bootstrap.InverseCDF(prob);
+            
+            lock (_bootstraplock)
+            {
+                double[] randyPacket = rp.NextRandomSequence(dist.SampleSize);
```

Comment:
can we stop using randypacket, please? this is completely unintuitive. it seems more helpful to call this arrayOfRandomNumbers